src/pages/Appointments/AppointmentsPage.tsx

```diff
@@ -749,9 +749,16 @@ export default function AppointmentsPage({ resourceType, resourceId }: Props) {
                 ) >= 31
               }
               onClick={() => {
-                navigate(`/facility/${facilityId}/appointments/print`, {
-                  query: qParams,
-                });
+                navigate(
+                  resourceType === SchedulableResourceType.HealthcareService
+                    ? `/facility/${facilityId}/services/${resourceId}/appointments/print`
+                    : resourceType === SchedulableResourceType.Location
+                      ? `/facility/${facilityId}/locations/${resourceId}/appointments/print`
+                      : `/facility/${facilityId}/appointments/print`,
```

Comment:
the only edge case is for practitioners, so you could simply check for practitioner first, and in the else directly return the other url. skips one additional if condition.

Reply:
you could simplify this even further by simply adding navigating to `"appointments/print"` instead. no if conditions that way. would end up using relative navigate

Reply:
91ee9c1ab050f17c664ca2d2447685f20c728992 works?